pymatviz/ptable.py

```diff
@@ -910,18 +915,212 @@ def ptable_hists(
+        ax.text(
+            *symbol_pos,
+            symbol_text(element)
+            if callable(symbol_text)
+            else symbol_text.format(elem=element),
+            ha="center",
+            va="center",
+            transform=ax.transAxes,
+            **symbol_kwargs,
+        )
+        ax.axis("on")  # re-enable axes of elements that exist
+
+        hist_data = data.get(symbol, [])
+        if anno_kwds:
+            defaults = dict(
+                text=lambda hist_vals: si_fmt_int(len(hist_vals)),
+                xy=(0.8, 0.8),
+                xycoords="axes fraction",
+                fontsize=8,
+                horizontalalignment="center",
+                verticalalignment="center",
+            )
+            if callable(anno_kwds):
+                annotation = anno_kwds(hist_data)
+            else:
+                annotation = anno_kwds
+                anno_text = anno_kwds.get("text")
+                if isinstance(anno_text, dict):
+                    anno_text = anno_text.get(symbol)
+                elif callable(anno_text):
+                    anno_text = anno_text(hist_data)
+                annotation["text"] = anno_text
+            ax.annotate(**(defaults | annotation))
+        if hist_data:
+            if callable(scatter_kwds):
+                hist_kwargs = scatter_kwds(hist_data)
+            else:
+                hist_kwargs = scatter_kwds
+            _n, bins_array, patches = ax.hist(
+                hist_data, bins=bins, log=log, range=x_range, **(hist_kwargs or {})
+            )
+            if x_range:
+                ax.set_xlim(x_range)
+            x_min, x_max = math.floor(min(bins_array)), math.ceil(max(bins_array))
+            x_ticks = list(x_range or [x_min, x_max])
+            if x_ticks[0] is None:
+                x_ticks[0] = x_min
+            if x_ticks[1] is None:
+                x_ticks[1] = x_max
+            if x_min < 0 < x_max:
+                # make sure we always show a mark at 0
+                x_ticks.insert(1, 0)
+
+            if cmap:
+                for patch, x_val in zip(patches, bins_array[:-1]):
+                    plt.setp(patch, "facecolor", cmap(norm(x_val)))
+            ax.set_xticks(x_ticks)
+            ax.tick_params(labelsize=8, direction="in")
+        else:  # disable ticks for elements without data
+            ax.set_xticks([])
+        ax.set_yticks([])  # disable y ticks for all elements
+
+        for side in ("left", "right", "top"):
+            ax.spines[side].set_visible(b=False)
+        # Hide y ticks
         ax.tick_params(axis="y", which="both", length=0)
 
-    # add colorbar
+    # Add colorbar
     if isinstance(cmap, Colormap):
         cbar_ax = fig.add_axes(cbar_coords)
         _cbar = fig.colorbar(
             plt.cm.ScalarMappable(norm=norm, cmap=cmap),
             cax=cbar_ax,
             **{"orientation": "horizontal"} | (cbar_kwds or {}),
         )
-        # set color bar title
```

Comment:
I personally prefer the `matplotlib` ["official"](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.colorbar.html) name of `colorbar` (similar to `colormap` instead of `color map`) just for consistency here. For example:

> matplotlib.pyplot.colorbar(mappable=None, cax=None, ax=None, **kwargs)[[source]](https://github.com/matplotlib/matplotlib/blob/v3.8.2/lib/matplotlib/pyplot.py#L2312-L2327)
Add a colorbar to a plot.

Reply:
the thing is colorbar isn't actually a word and comments are natural language

Reply:
Yes I agree neither _color bar_ nor _color map_ is naturally a word, but in the context of `matplotlib` package, the official documentations use `colorbar/colormap` in comments and docstrings instead of `color bar/color map` anyway, for example the `matplotlib.pyplot.colorbar` [documentation](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.colorbar.html) there is:

> Parameters:
> `mappable`
> The [matplotlib.cm.ScalarMappable](https://matplotlib.org/stable/api/cm_api.html#matplotlib.cm.ScalarMappable) (i.e., [AxesImage](https://matplotlib.org/stable/api/image_api.html#matplotlib.image.AxesImage), [ContourSet](https://matplotlib.org/stable/api/contour_api.html#matplotlib.contour.ContourSet), etc.) described by this **colorbar**. This argument is mandatory for the [Figure.colorbar](https://matplotlib.org/stable/api/figure_api.html#matplotlib.figure.Figure.colorbar) method but optional for the [pyplot.colorbar](https://matplotlib.org/stable/api/_as_gen/matplotlib.pyplot.colorbar.html#matplotlib.pyplot.colorbar) function, which sets the default to the current image.
> 
> Note that one can create a [ScalarMappable](https://matplotlib.org/stable/api/cm_api.html#matplotlib.cm.ScalarMappable) "on-the-fly" to generate **colorbars** not attached to a previously drawn artist, e.g.

> `fig.colorbar(cm.ScalarMappable(norm=norm, cmap=cmap), ax=ax)`

I personally want to keep consistency with upstream `matplotlib` wordings, but I would be happy to revert these changes if you prefer not to change them here.

Reply:
> I personally want to keep consistency with upstream matplotlib wording

sure, no need to revert